python/lsst/ts/Dome/mock_controller.py

```diff
@@ -240,6 +224,22 @@ async def status(self):
         data = yaml.safe_dump(reply, default_flow_style=None)
         await self.write("OK:\n" + data)
 
+    async def determine_statuses(self):
+        """Determine the status of the lower level components.
+        """
+        self.log.debug(f"self.current_tai = {self.current_tai}")
+        await self.amcs.determine_status(self.current_tai)
+        await self.apscs.determine_status(self.current_tai)
+        await self.lcs.determine_status(self.current_tai)
+        await self.lwscs.determine_status(self.current_tai)
+        await self.moncs.determine_status(self.current_tai)
+        await self.thcs.determine_status(self.current_tai)
+
+    async def determine_current_tai(self):
```

Comment:
Please consider deleting this method and just calling determine_statuses with the desired (presumably current) time.

If you want to record the time at which status was most recently computed (I have no idea if it's useful), you can do that as the first line of determine_statuses.

Reply:
The reason that I introduced this method is to be able to mock in in the unit tests so I can control the time more exactly. If the determine_statuses method determines the status then it will be much harder to mock the time. So I prefer to keep it.